encode.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"os"
+
+	vegeta "github.com/tsenart/vegeta/lib"
+)
+
+const (
+	encodingCSV  = "csv"
+	encodingGob  = "gob"
+	encodingJSON = "json"
+)
+
+type decoderFunc func(io.Reader) vegeta.Decoder
+type encoderFunc func(io.Writer) vegeta.Encoder
+
+func encodeCmd() command {
+	var (
+		fs     = flag.NewFlagSet("encode", flag.ExitOnError)
+		from   = fs.String("from", "gob", "Input decoding [csv, gob, json]")
+		to     = fs.String("to", "json", "Output encoding [csv, gob, json]")
+		output = fs.String("output", "", "Output file")
+	)
+
+	fs.Usage = func() {
+		fmt.Println("Usage: vegeta encode [flags] [<file>...]")
+		fs.PrintDefaults()
+	}
+
+	return command{
+		fs,
+		func(args []string) error {
+			fs.Parse(args)
+			return encode(*from, *to, *output, args...)
+		},
+	}
+}
+
+func encode(from, to, output string, inputs ...string) error {
+	var decFn decoderFunc
+
+	switch from {
+	case encodingCSV:
+		decFn = vegeta.NewCSVDecoder
+	case encodingJSON:
+		decFn = vegeta.NewJSONDecoder
+	default:
+		// Gob is our default decode format to play nicely in pipes with attack.
+		decFn = vegeta.NewDecoder
+	}
+
+	decs := []vegeta.Decoder{}
+
+	if len(inputs) > 0 {
```

Comment:
We have some simpler code to handle this already. i.e. https://github.com/tsenart/vegeta/blob/master/report.go#L33

Because the `file` function returns `os.Stdin` for the string `stdin`. This block of code can become:

```
	srcs := make([]vegeta.Decoder, len(files))
	for i, f := range inputs {
		in, err := file(f, false)
		if err != nil {
			return err
		}
		defer in.Close()
		srcs[i] = vegeta.NewDecoder(in)
	}
	dec := vegeta.NewRoundRobinDecoder(srcs...)
```

Reply:
Yeah, `file` is slightly conflated. We can take out the file creation part from it and use `os.Create` directly where needed, thus scoping the symbolic resolution to opening files only, not creating (since you don't create `stdout` or `stdin`).

```go
func open(file string) (*os.File, error) {
    switch file {
    case "stdout":
        return os.Stdout, nil
    case "stdin":
        return os.Stdin, nil
    default:
        return os.Open(file)
}
```